npc/events/christmas_2006.txt

```diff
@@ -0,0 +1,4165 @@
+			next;
+			mes "[Ohholy]";
+			mes "......................What?";
+			next;
+			mes "[Ohholy]";
+			mes "Ah! Ah! Hahahaha.";
+			mes "No, I didn't. How can I forget about it? Of course I remember!";
+			next;
+			mes "[Ohholy]";
+			mes "...Hum... by the way, why did I promise to make you the ^0000ffBlessed powder for purification^000000?";
+			next;
+			mes "["+ strcharinfo(0) +"]";
+			mes "(She completely forgot it!)";
+			mes "...Well..";
+			next;
+			switch(select("To drive all the devil away", "To purify the monster's egg", "To get rid of headache")) {
+				case 1:
+					mes "[Ohholy]";
+					mes "......Haha? For that reason?";
+					next;
+					mes "[Ohholy]";
+					mes "You don't need ^0000ffBlessed powder for purification^000000 to drive away the devil!";
+					next;
+					mes "[Ohholy]";
+					mes "I'll crack it down for myself !";
+					close;
+				case 2:
+					mes "[Ohholy]";
+					mes "Yes, that's it. I remember. Hehe.";
+					next;
+					mes "[Ohholy]";
+					mes "Then... let me see,  seems like you've got all stuffs for the powder, I'll make it right away.";
+					next;
+					mes "[Ohholy]";
+					mes "First, I have to break the broken diamonds and make the powder of it! It requires know-how. I learned it from a smith.";
+					specialeffect EF_REPAIRWEAPON;
+					emotion ET_HUK;
+					next;
+					mes "[Ohholy]";
+					mes "You'd better pretend to be surprised with it";
+					next;
+					mes "[Ohholy]";
+					mes "Next, pour water into a large pot and boil slowly. Melt red and yellow gemston perfectly.";
+					specialeffect EF_DEMONSTRATION;
+					next;
+					mes "[Ohholy]";
+					mes "Mix the powder of the diamond with the holy water made from gemstones and dry it in the sun! Ok, it's done! Here is ^0000ffBlessed powder for purification^000000!!! ";
+					specialeffect EF_ICECRASH;
+					next;
+					mes "[Ohholy]";
+					mes "Now, you go purify the monster's egg. I wish good luck to you.";
+					emotion ET_SMILE;
+					next;
+					mes "-You've got^0000ffBlessed powder for purification!^000000";
+					delitem 733, 1; // Crystal_Jewel___
+					delitem 523, 3; // Holy_Water
+					delitem 716, 1; // Red_Gemstone
+					delitem 715, 1; // Yellow_Gemstone
+					06christ_santa = 3;
+					close;
+				case 3:
+					mes "[Ohholy]";
+					mes "You shouldn't have drunk too much!";
+					next;
+					mes "[Ohholy]";
+					mes "To relieve the hangover, we don't need ^0000ffBlessed powder for purification^000000.Just do as I say, Ok? Relax and stand there.";
+					next;
+					mes "[Ohholy]";
+					mes "Ok! Here we go!";
+					mes "'Father, you always cure our injury! The first movement!'";
+					specialeffect EF_GLORIA;
+					specialeffect2 EF_STUNATTACK;
+					emotion ET_HUK, playerattached();
+					percentheal -30, 0;
+					next;
+					mes "["+ strcharinfo(0) +"]";
+					mes "Oh, no!!Stop it! I have a real bad headache!!!";
+					next;
+					mes "[Ohholy]";
+					mes "If you listen all 74th
```

Comment:
`.@name$ = strcharinfo(0);` to avoid the redundant command call

Reply:
Applied suggested fix.